### ISSUE TITLE
Autoupdate doesn't seem to work

### ISSUE THREAD

**Issue by mgol** [sindresorhus/caprine#190]
Caprine 2.1.0 has been out for 6 days and my Caprine 2.0.0 macOS installation didn't get updated. How soon is the update supposed to get picked up? I'm not sure if it works.

**Comment by sindresorhus** [comment]
It should update right away. You should have gotten the new version after a app restart.

Can you share the output of `~/Library/Logs/Caprine/log.log`?

**Comment by mgol** [comment]
```
cat ~/Library/Logs/Caprine/log.log
[2017-04-03 12:11:11:0386] [info] Checking for update
[2017-04-03 12:11:12:0758] [info] Update for version 2.0.0 is not available (latest version: 2.0.0)
[2017-04-07 15:32:26:0180] [info] Checking for update
[2017-04-07 15:32:28:0616] [info] Update for version 2.0.0 is not available (latest version: 2.0.0)
[2017-04-07 15:40:28:0566] [info] Checking for update
[2017-04-07 15:40:30:0371] [info] Update for version 2.0.0 is not available (latest version: 2.0.0)
[2017-04-10 20:59:40:0512] [info] Checking for update
[2017-04-10 20:59:44:0321] [info] Update for version 2.0.0 is not available (latest version: 2.0.0)
[2017-04-14 15:36:47:0587] [info] Checking for update
[2017-04-14 15:36:49:0815] [info] Update for version 2.0.0 is not available (latest version: 2.0.0)
[2017-04-14 17:06:39:0867] [info] Checking for update
[2017-04-14 17:06:42:0739] [info] Update for version 2.0.0 is not available (latest version: 2.0.0)
[2017-04-24 12:27:38:0348] [info] Checking for update
[2017-04-24 12:27:39:0910] [info] Found version 2.1.0 (url: https://github.com/sindresorhus/caprine/releases/download/v2.1.0/caprine-2.1.0-mac.zip)
[2017-04-24 12:27:39:0911] [info] Downloading update from https://github.com/sindresorhus/caprine/releases/download/v2.1.0/caprine-2.1.0-mac.zip
[2017-04-24 12:30:01:0940] [info] New version 2.1.0 has been downloaded
[2017-04-24 13:00:52:0002] [info] Checking for update
[2017-04-24 13:00:53:0436] [info] Found version 2.1.0 (url: https://github.com/sindresorhus/caprine/releases/download/v2.1.0/caprine-2.1.0-mac.zip)
[2017-04-24 13:00:53:0437] [info] Downloading update from https://github.com/sindresorhus/caprine/releases/download/v2.1.0/caprine-2.1.0-mac.zip
```

I got intrigued why it's firing up 2.0.0 since it claims it downloaded 2.1.0 and it seems if you open it from a Dock (even after quitting the app) it opens the same version as the last time. :/

I quit Caprine again and launched it from the Launchpad this time and it opened 2.1.0. Now even if I quit it again and open from the Dock it opens 2.1.0 correctly.

I wonder why it's required to open not from a Dock at least once between updates...

I'm on latest Sierra (10.12.4 (16E195)) if it matters.

**Comment by sindresorhus** [comment]
Are you sure you don't have multiple Caprine installations? Try right-clicking the Caprine dock icon, select Options, and "Show in Finder", and see where it's located.

**Comment by mgol** [comment]
"Show in Finder" shows the one in `/Applications/Caprine.app` (note, though, that I have it "fixed" for the current upgrade as it opens 2.1.0 now for me after I opened it from the Launchpad once). `mdfind Caprine.app` only lists `/Applications/Caprine.app`, the Finder search for `caprine.app` also only finds this one instance.

**Comment by sindresorhus** [comment]
@develar Any ideas?

**Comment by develar** [comment]
Please open Console (system macOS app) — do you see any related messages (search by Caprine  or Squirrel.Mac)?

**Comment by mgol** [comment]
In which logs should I search? (see the screenshot) I don't see anything under Devices.

I've tried it again - I installed 2.0.0, removed Caprine from the Dock, opened once from Launchpad & re-pinned to the Dock, it opened the just installed 2.0.0. I then waited for log.log to indicate 2.1.0 has been downloaded, closed the app and opened from the Dock this time and it had 2.1.0.

I'm not sure how to reproduce the initial issue. I may try again when a next Caprine update comes out, maybe it only breaks if the previous version is installed when there's nothing newer yet.

**Comment by develar** [comment]
@mgol Yes, in the `mgol-mbpro` category.

I will try to reproduce.

**Comment by mgol** [comment]
I have access to a Windows 10 machine which still has Caprine 2.0.0 that hasn't updated to 2.1.0 so this problem seems to be cross-platform.

What can I check on this Windows 10 laptop to help debug the issue?

**Comment by sindresorhus** [comment]
Just for reference. Updating works for me on both macOS and Windows 10.

**Comment by sindresorhus** [comment]
@mgol The logs would be helpful: `%USERPROFILE%\AppData\Roaming\Caprine\log.log`

**Comment by mgol** [comment]
It's weird. I've tried restarting the app on Windows 10 some a few days ago (after reporting this issue) and it opened 2.0.0 again. I've tried it now again and it did open 2.1.0. The logs are as follows:
```
[2017-04-05 20:12:27:0874] [info] Checking for update
[2017-04-05 20:12:31:0019] [info] Update for version 2.0.0 is not available (latest version: 2.0.0)
[2017-04-05 20:16:15:0995] [info] Checking for update
[2017-04-05 20:16:19:0300] [info] Update for version 2.0.0 is not available (latest version: 2.0.0)
[2017-04-11 21:41:14:0652] [info] Checking for update
[2017-04-11 21:41:17:0438] [info] Update for version 2.0.0 is not available (latest version: 2.0.0)
[2017-04-12 21:32:18:0791] [info] Checking for update
[2017-04-12 21:32:22:0082] [info] Update for version 2.0.0 is not available (latest version: 2.0.0)
[2017-04-29 18:01:09:0838] [info] Checking for update
[2017-04-29 18:01:12:0935] [info] Found version 2.1.0 (url: https://github.com/sindresorhus/caprine/releases/download/v2.1.0/caprine-setup-2.1.0.exe)
[2017-04-29 18:01:12:0941] [info] Downloading update from https://github.com/sindresorhus/caprine/releases/download/v2.1.0/caprine-setup-2.1.0.exe
[2017-04-29 18:01:52:0606] [info] New version 2.1.0 has been downloaded to C:\Users\Joanna\AppData\Local\Temp\up-CGRKJQ\caprine-setup-2.1.0.exe
[2017-05-01 13:35:34:0831] [info] Auto install update on quit
[2017-05-01 13:35:52:0973] [info] Checking for update
[2017-05-01 13:35:56:0681] [info] Update for version 2.1.0 is not available (latest version: 2.1.0, downgrade is disallowed.
```

Feel free to close if you can't reproduce; I may open another issue if it happens with the next updates (I'll make sure to pay more attention to the time at which I check various operations and will compare it to the logs).

**Comment by sindresorhus** [comment]
I just did a new update. Can you open Caprine and check your logs?

**Comment by sindresorhus** [comment]
I just opened Caprine and it encountered an error. After opening it a second time, it seems to work.

```
[2017-05-01 21:16:14:0262] [info] Checking for update
[2017-05-01 21:16:41:0788] [error] Error: Error: net::ERR_EMPTY_RESPONSE
    at CancellationToken.createPromise (/Applications/Caprine.app/Contents/Resources/app.asar/node_modules/electron-builder-http/src/CancellationToken.ts:51:5)
    at ElectronHttpExecutor.doApiRequest (/Applications/Caprine.app/Contents/Resources/app.asar/node_modules/electron-updater/src/electronHttpExecutor.ts:44:30)
    at ElectronHttpExecutor.request (/Applications/Caprine.app/Contents/Resources/app.asar/node_modules/electron-builder-http/src/httpExecutor.ts:78:17)
    at request (/Applications/Caprine.app/Contents/Resources/app.asar/node_modules/electron-builder-http/src/httpExecutor.ts:213:38)
    at /Applications/Caprine.app/Contents/Resources/app.asar/node_modules/electron-updater/src/GitHubProvider.ts:65:22
From previous event:
    at GitHubProvider.getLatestVersion (/Applications/Caprine.app/Contents/Resources/app.asar/node_modules/electron-updater/out/GitHubProvider.js:123:11)
    at /Applications/Caprine.app/Contents/Resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:192:38
From previous event:
    at MacUpdater.doCheckForUpdates (/Applications/Caprine.app/Contents/Resources/app.asar/node_modules/electron-updater/out/AppUpdater.js:263:11)
    at /Applications/Caprine.app/Contents/Resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:177:25
    at Generator.next (<anonymous>)
    at runCallback (timers.js:651:20)
    at tryOnImmediate (timers.js:624:5)
    at processImmediate [as _immediateCallback] (timers.js:596:5)
From previous event:
    at MacUpdater._checkForUpdates (/Applications/Caprine.app/Contents/Resources/app.asar/node_modules/electron-updater/out/AppUpdater.js:222:11)
    at MacUpdater.checkForUpdates (/Applications/Caprine.app/Contents/Resources/app.asar/node_modules/electron-updater/src/AppUpdater.ts:159:35)
    at Object.<anonymous> (/Applications/Caprine.app/Contents/Resources/app.asar/index.js:142:14)
    at Object.<anonymous> (/Applications/Caprine.app/Contents/Resources/app.asar/index.js:194:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Object.<anonymous> (/Applications/Caprine.app/Contents/Resources/electron.asar/browser/init.js:171:8)
    at Object.<anonymous> (/Applications/Caprine.app/Contents/Resources/electron.asar/browser/init.js:173:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.runMain (module.js:605:10)
    at run (bootstrap_node.js:424:7)
    at startup (bootstrap_node.js:147:9)
[2017-05-01 21:17:43:0699] [info] Checking for update
[2017-05-01 21:17:47:0242] [info] Found version 2.2.0 (url: https://github.com/sindresorhus/caprine/releases/download/v2.2.0/caprine-2.2.0-mac.zip)
[2017-05-01 21:17:47:0243] [info] Downloading update from https://github.com/sindresorhus/caprine/releases/download/v2.2.0/caprine-2.2.0-mac.zip
[2017-05-01 21:18:17:0247] [info] New version 2.2.0 has been downloaded
[2017-05-01 21:19:14:0612] [info] Checking for update
[2017-05-01 21:19:19:0595] [info] Update for version 2.2.0 is not available (latest version: 2.2.0, downgrade is disallowed.
```

**Comment by mgol** [comment]
I restarted the app, waited a little & restarted it again and it worked fine now; here are the Windows logs:
```
[2017-04-05 20:12:27:0874] [info] Checking for update
[2017-04-05 20:12:31:0019] [info] Update for version 2.0.0 is not available (latest version: 2.0.0)
[2017-04-05 20:16:15:0995] [info] Checking for update
[2017-04-05 20:16:19:0300] [info] Update for version 2.0.0 is not available (latest version: 2.0.0)
[2017-04-11 21:41:14:0652] [info] Checking for update
[2017-04-11 21:41:17:0438] [info] Update for version 2.0.0 is not available (latest version: 2.0.0)
[2017-04-12 21:32:18:0791] [info] Checking for update
[2017-04-12 21:32:22:0082] [info] Update for version 2.0.0 is not available (latest version: 2.0.0)
[2017-04-29 18:01:09:0838] [info] Checking for update
[2017-04-29 18:01:12:0935] [info] Found version 2.1.0 (url: https://github.com/sindresorhus/caprine/releases/download/v2.1.0/caprine-setup-2.1.0.exe)
[2017-04-29 18:01:12:0941] [info] Downloading update from https://github.com/sindresorhus/caprine/releases/download/v2.1.0/caprine-setup-2.1.0.exe
[2017-04-29 18:01:52:0606] [info] New version 2.1.0 has been downloaded to C:\Users\Joanna\AppData\Local\Temp\up-CGRKJQ\caprine-setup-2.1.0.exe
[2017-05-01 13:35:34:0831] [info] Auto install update on quit
[2017-05-01 13:35:52:0973] [info] Checking for update
[2017-05-01 13:35:56:0681] [info] Update for version 2.1.0 is not available (latest version: 2.1.0, downgrade is disallowed.
[2017-05-01 17:28:04:0098] [info] Checking for update
[2017-05-01 17:28:07:0843] [info] Found version 2.2.0 (url: https://github.com/sindresorhus/caprine/releases/download/v2.2.0/caprine-setup-2.2.0.exe)
[2017-05-01 17:28:07:0848] [info] Downloading update from https://github.com/sindresorhus/caprine/releases/download/v2.2.0/caprine-setup-2.2.0.exe
[2017-05-01 17:28:31:0401] [info] New version 2.2.0 has been downloaded to C:\Users\Joanna\AppData\Local\Temp\up-eqjOPB\caprine-setup-2.2.0.exe
[2017-05-01 17:29:00:0827] [info] Auto install update on quit
[2017-05-01 17:29:14:0310] [info] Checking for update
[2017-05-01 17:29:18:0058] [info] Update for version 2.2.0 is not available (latest version: 2.2.0, downgrade is disallowed.
```

Does it poll for updates periodically while it's running or only just at the app start?

**Comment by develar** [comment]
> Error: Error: net::ERR_EMPTY_RESPONSE 

I have filed as #1522
